test/hummingbot/connector/exchange/latoken/test_latoken_api_order_book_data_source.py

```diff
@@ -0,0 +1,548 @@
+        self.assertEqual(0, len(result))
+
+    @aioresponses()
+    def test_get_snapshot_successful(self, mock_api):
+        path_url = f"{CONSTANTS.BOOK_PATH_URL}/{self.base_asset}/{self.quote_asset}"
+        url = web_utils.public_rest_url(path_url=path_url, domain=self.domain)
+        regex_url = re.compile(f"^{url}".replace(".", r"\.").replace("?", r"\?"))
+
+        mock_api.get(regex_url, body=json.dumps(self._snapshot_response()))
+
+        result: Dict[str, Any] = self.async_run_with_timeout(
+            self.data_source.get_snapshot(self.trading_pair)
+        )
+
+        self.assertEqual(self._snapshot_response(), result)
+
+    @aioresponses()
+    def test_get_snapshot_catch_exception(self, mock_api):
+        path_url = f"{CONSTANTS.BOOK_PATH_URL}/{self.base_asset}/{self.quote_asset}"
+        url = web_utils.public_rest_url(path_url=path_url, domain=self.domain)
+        regex_url = re.compile(f"^{url}".replace(".", r"\.").replace("?", r"\?"))
+
+        mock_api.get(regex_url, status=400)
+        with self.assertRaises(IOError):
+            self.async_run_with_timeout(
+                self.data_source.get_snapshot(self.trading_pair)
+            )
+
+    @aioresponses()
+    def test_get_new_order_book(self, mock_api):
+        path_url = f"{CONSTANTS.BOOK_PATH_URL}/{self.base_asset}/{self.quote_asset}"
+        url = web_utils.public_rest_url(path_url=path_url, domain=self.domain)
+        regex_url = re.compile(f"^{url}".replace(".", r"\.").replace("?", r"\?"))
+
+        mock_response: Dict[str, Any] = {
+            "ask": [
+                {
+                    "price": "123.321",
+                    "quantity": "0.12",
+                    "cost": "14.79852",
+                    "accumulated": "14.79852"
+                },
+                {
+                    "price": "123.321",
+                    "quantity": "0.12",
+                    "cost": "14.79852",
+                    "accumulated": "14.79852"
+                }
+            ],
+            "bid": [
+                {
+                    "price": "123.321",
+                    "quantity": "0.12",
+                    "cost": "14.79852",
+                    "accumulated": "14.79852"
+                },
+                {
+                    "price": "123.321",
+                    "quantity": "0.12",
+                    "cost": "14.79852",
+                    "accumulated": "14.79852"
+                }
+            ],
+            "totalAsk": "...",
+            "totalBid": "..."
+        }
+        mock_api.get(regex_url, body=json.dumps(mock_response))
+
+        result: OrderBook = self.async_run_with_timeout(
+            self.data_source.get_new_order_book(self.trading_pair)
+        )
+
+        self.assertGreater(result.snapshot_uid, 1)
+
+    @patch("aiohttp.ClientSession.ws_connect", new_callable=AsyncMock)
+    def test_listen_for_subscriptions_subscribes_to_trades_and_order_diffs(self, ws_connect_mock):
+        import aiohttp
```

Comment:
We try to avoid local imports whenever possible

Reply:
removed this and put import on top, it is necessary for the enums we use